### PR TITLE
Fine Tuning

### PR DESCRIPTION
**Pull request by indexxy**
This is just a first step.
Many details can still be implemented, like saving the training data along with metadata containing the network's layers, activation functions and training dataset's name.

### CODE REVIEW

**Comment by volf52** [comment]
I'll make the changes tomorrow morning and merge.

- Saved: Object containing weights, biases and architecture. Should save the training set name as it makes it too rigid. 

- The file name will be "{datasetName}_{time}.npz". NPZ format to save the object in compressed binary form.

- The fine-tuning/save option should be in the training function and not the part of the constructor. This also means I will move the weight init from constructor to train func.

- Some changes will be made to ensure best weights are saved based on training/validation accuracy (not test accuracy).

- Separate function for test accuracy.

**Comment by volf52** [comment]
Thanks for starting this btw.

**Comment by volf52** [comment]
Finetuning ready
